extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3Utils.java

```diff
@@ -70,8 +71,13 @@ public boolean apply(Throwable e)
         return false;
       } else if (e instanceof IOException) {
         return true;
+      } else if (e instanceof java.lang.InterruptedException) {
+        Thread.interrupted(); // Clear flag and retry
+        return true;
```

Comment:
if it gets interrupted it means someone is asking the thread to shut down, so we should bail and not retry.

Reply:
> if it gets interrupted it means someone is asking the thread to shut down, so we should bail and not retry.

Changed to not retry by return false.